share/sysbuild/cmake/modules/sysbuild_extensions.cmake

```diff
@@ -351,6 +368,93 @@ function(ExternalZephyrProject_Cmake)
   import_kconfig(CONFIG_ ${BINARY_DIR}/zephyr/.config TARGET ${ZCMAKE_APPLICATION})
 endfunction()
 
+# Usage:
+#   ExternalZephyrProject_Group(<name>
```

Comment:
@tejlmand: https://github.com/zephyrproject-rtos/zephyr/pull/57884#discussion_r1200509550

> Not sure about the `_Group` name here.
> 
> The `ExternalZephyrProject_Group()` seems to more handle properties and not be doing an explicit step as described: https://cmake.org/cmake/help/latest/module/ExternalProject.html#explicit-step-management
> 
> That's a difference to the `ExternalZephyrProject_CMake()` function which actually carries out a dedicate step, which is the CMake configure step, but with the slight difference that the step is executed at sysbuild CMake configure time and not sysbuild build time (which would be the normal behavior of an external step).

I don't see how this is relevant. What you linked is documentation for `ExternalProject_Add_Step()`, `ExternalProject_Add_StepTargets()`, and `ExternalProject_Add_StepDependencies()`.

Could one argue that `ExternalZephyrProject_Add()` also doesn't carry out a dedicated step, because it just adds a target?

If you see a different problem with the function name, then I can change it, but I want to keep it concise.

Reply:
> I don't see how this is relevant.

ExternalProject is a CMake module for incorporating other projects into current project.
This is done by `_Add` command, where sysbuild provides a function dedicated for Zephyr projects.

The `_Group()` doesn't seem to require an already created external project, and it doesn't create any external projects of it's own, hence it doesn't seem to be belonging inside the `ExternalProject` or `ExternalZephyrProject` domain.
It could potentially belong there if it was operating on a dedicate external project target, as example setting properties on such target, but then the function name should reflect such use.

Of course there could potentially be commands which can be argued to have a `ExternalZephyrProject_<something>` name without carrying out a dedicated step, but in this case I don't see enough justification as to why `_Group` is such a command, especially as it doesn't work directly on a created project, nor does it carry out a step on such target.

Reply:
Those are fair points. I'm fine with renaming the function to something simple like `define_group()`.